WikipediaUnitTests/Code/WMFBlacklistTests.m

```diff
@@ -0,0 +1,53 @@
+//
+//  WMFBlacklistTests.m
+//  Wikipedia
+//
+//  Created by Corey Floyd on 1/21/16.
+//  Copyright © 2016 Wikimedia Foundation. All rights reserved.
+//
+
+#import <XCTest/XCTest.h>
+#import "WMFAsyncTestCase.h"
+#import "WMFRelatedSectionBlackList.h"
+#import "MWKTitle.h"
+
+@interface WMFRelatedSectionBlackList (WMFTesting)
+
++ (instancetype)loadFromDisk;
+
+@end
+
+
+@interface WMFBlacklistTests : WMFAsyncTestCase
+
+@end
+
+@implementation WMFBlacklistTests
+
+- (void)tearDown {
+    WMFRelatedSectionBlackList* bl = [[WMFRelatedSectionBlackList alloc] init];
+    [bl removeAllEntries];
+    [bl save];
+    [super tearDown];
+}
+
+- (void)testPersistsToDisk {
+    PushExpectation();
+    MWKTitle* title                = [[MWKTitle alloc] initWithString:@"some-title" site:[MWKSite siteWithCurrentLocale]];
+    WMFRelatedSectionBlackList* bl = [[WMFRelatedSectionBlackList alloc] init];
+    [bl addEntry:title];
+    [bl save].then(^(){
+        [self popExpectationAfter:nil];
+    }).catch(^(NSError* error){
+        XCTFail(@"Error callback erroneously called with error %@", error);
+    });
+    WaitForExpectations();
+
+    bl = [WMFRelatedSectionBlackList loadFromDisk];
+    MWKTitle* first = [[bl entries] firstObject];
+
+    XCTAssertTrue([title isEqual:first],
```

Comment:
So… hamcrest is terrible as I have said before - that why i don't use that in general - and you can't add quick/nimble without ruining code completion and showing errors (since you have to use @import to get it to work). So I choose to use the built in stuff instead.